firmware/components/micropython/esp32/modframebuffer.c

```diff
@@ -687,6 +889,381 @@ static mp_obj_t framebuffer_default_fill_color(mp_uint_t n_args, const mp_obj_t
+/* Arguments: window (optional) */
+
+static MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN( framebuffer_popMatrix_obj,           0, 1, framebuffer_popMatrix);
+/* Arguments: window (optional) */
+
+static MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN( framebuffer_clearMatrix_obj,         0, 2, framebuffer_clearMatrix);
+/* Arguments: window (optional) */
+
+static MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN( framebuffer_matrixSize_obj,          0, 1, framebuffer_matrixSize);
+/* Arguments: window (optional) */
+
+static MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN( framebuffer_getMatrix_obj,           0, 1, framebuffer_getMatrix);
+/* Arguments: window (optional) */
+
+static MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN( framebuffer_setMatrix_obj,           1, 2, framebuffer_setMatrix);
+/* Arguments: window (optional), matrix */
+
+static MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN( framebuffer_transformPoint_obj,      1, 2, framebuffer_transformPoint);
+/* Arguments: window (optional), x, y */
+
+static MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN( framebuffer_translate_obj,           2, 3, framebuffer_translate);
+/* Arguments: window (optional), x, y */
+
+static MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN( framebuffer_rotate_obj,              1, 2, framebuffer_rotate);
+/* Arguments: window (optional), angle */
+
+static MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN( framebuffer_scale_obj,               2, 3, framebuffer_scale);
+/* Arguments: window (optional), xscale, yscale */
+
+	// {MP_ROM_QSTR( MP_QSTR_pushMatrix                    ), MP_ROM_PTR( &framebuffer_pushMatrix_obj           )}, //Push the current matrix onto the stack
+	// {MP_ROM_QSTR( MP_QSTR_popMatrix                     ), MP_ROM_PTR( &framebuffer_popMatrix_obj            )}, //Pop the top matrix off the stack
+	// {MP_ROM_QSTR( MP_QSTR_clearMatrix                   ), MP_ROM_PTR( &framebuffer_clearMatrix_obj          )}, //Clear the matrix stack
+	// {MP_ROM_QSTR( MP_QSTR_matrixSize                    ), MP_ROM_PTR( &framebuffer_matrixSize_obj           )}, //Get the size of the matrix stack
+	// {MP_ROM_QSTR( MP_QSTR_getMatrix                     ), MP_ROM_PTR( &framebuffer_getMatrix_obj            )}, //Get the current matrix
+	// {MP_ROM_QSTR( MP_QSTR_setMatrix                     ), MP_ROM_PTR( &framebuffer_setMatrix_obj            )}, //Set the current matrix
+	// {MP_ROM_QSTR( MP_QSTR_transformPoint                ), MP_ROM_PTR( &framebuffer_transformPoint_obj       )}, //Transform a point by the current matrix
+	// {MP_ROM_QSTR( MP_QSTR_translate                     ), MP_ROM_PTR( &framebuffer_translate_obj            )}, //Translate (move) the canvas
+	// {MP_ROM_QSTR( MP_QSTR_rotate                        ), MP_ROM_PTR( &framebuffer_rotate_obj               )}, //Rotate the canvas around the origin
+	// {MP_ROM_QSTR( MP_QSTR_scale                         ), MP_ROM_PTR( &framebuffer_scale_obj                )}, //Scale the canvas
+#endif //CONFIG_G_MATRIX_ENABLE
+//End matrix stack

```

Comment:
Should these not be enabled?

Reply:
These should be removed as they are a duplicate of the set of commands defined later in the file